app/tests/e2e.test.ts

```diff
@@ -0,0 +1,141 @@
+import fetch from "node-fetch";
```

Comment:
I feel like there must be a way to do this sort of testing at a higher level. Is there a library or tool that supports contract based testing in jest?

Reply:
There is supertest, but it requires the app to be isolated and exported. We do not export app directly in node-raml-mocker, rather we export functions that pre-processes and starts the server on app. That part needs to be refactored if supertest needs to be used. 

One of the benefit of using fetch for integration testing is that our application is running as a docker image, then it is being queried using fetch, just like it would be interacted with in the real world. 

Pact seems to be popular for contract based testing with jest. But that mocks the server and then makes the requests. That may not be suitable for our case. 

https://www.mariedrake.com/post/contract-testing-with-pact-js-and-jest